@loicmathieu seems like you need to format the code:

```
2020-02-26T11:11:05.5002739Z [ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.3.2:check (check-imports) on project quarkus-mongodb-panache: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java -> [Help 1]
2020-02-26T11:11:05.5003468Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.3.2:check (check-imports) on project quarkus-mongodb-panache: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
```

Once that is done, I'll approve the PR

--- a/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
+++ b/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/binder/CommonQueryBinder.java
@@ -9,6 +9,8 @@ import java.time.format.DateTimeFormatter;
 import java.util.Date;
 import java.util.TimeZone;
 
+import org.bson.types.ObjectId;
+
 final class CommonQueryBinder {
 
     private static final String ISO_DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
@@ -43,6 +45,10 @@ final class CommonQueryBinder {
         if (value instanceof Instant) {
             Instant dateValue = (Instant) value;
             return "ISODate('" + ISO_DATE_FORMATTER.format(dateValue.atZone(ZoneOffset.UTC)) + "')";
+        }
+        if (value instanceof ObjectId) {
+            ObjectId objectId = (ObjectId) value;
+            return "ObjectId('" + objectId.toHexString() + "')";
         }
         return "'" + value.toString().replace("\\", "\\\\").replace("'", "\\'") + "'";
     }

--- a/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/BugResource.java
+++ b/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/BugResource.java
@@ -9,6 +9,7 @@ import java.util.Date;
 import javax.inject.Inject;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
@@ -77,6 +78,23 @@ public class BugResource {
         if (result == null) {
             return Response.status(404).build();
         }
+        return Response.ok().build();
+    }
+
+    @GET
+    @Path("7415")
+    public Response testForeignObjectId() {
+        LinkedEntity link = new LinkedEntity();
+        link.name = "toto";
+        link.persist();
+
+        LinkedEntity entity = new LinkedEntity();
+        entity.name = "tata";
+        entity.myForeignId = link.id;
+        entity.persist();
+
+        // we should be able to retrieve `entity` from the foreignId ...
+        LinkedEntity.find("myForeignId", link.id).firstResultOptional().orElseThrow(() -> new NotFoundException());
         return Response.ok().build();
     }
 }

--- a/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/LinkedEntity.java
+++ b/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/LinkedEntity.java
@@ -1,0 +1,10 @@
+package io.quarkus.it.mongodb.panache.bugs;
+
+import org.bson.types.ObjectId;
+
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+
+public class LinkedEntity extends PanacheMongoEntity {
+    public String name;
+    public ObjectId myForeignId;
+}

--- a/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongodbPanacheResourceTest.java
+++ b/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongodbPanacheResourceTest.java
@@ -329,4 +329,9 @@ class MongodbPanacheResourceTest {
 
         get("/bugs/6324/abstract").then().statusCode(200);
     }
+
+    @Test
+    public void testBug7415() {
+        get("/bugs/7415").then().statusCode(200);
+    }
 }

